### PR TITLE
Adding include that makes the intel compiler cry

### PR DESCRIPTION
**Pull request by sithhell**
Flyby:
    - using broadcast_apply to set updategroup indices
    - changing return value to avoid explicit get

### CODE REVIEW

File: src/communication/hpxreceiver.h [hunk 1]
```diff
@@ -62,6 +63,20 @@ class HPXReceiver : public hpx::components::simple_component_base<HPXReceiver<CA
         return ids;
     }
 
+    virtual ~HPXReceiver()
```
Comment (gentryx) [inline src/communication/hpxreceiver.h]: why did you move these functions up?

Comment (sithhell) [inline src/communication/hpxreceiver.h]: Ups, I played around fixing the issues I ran into. Nothing changed there, I merely forgot to undo those changes.

**Comment by gentryx** [comment]
Thanks! I'm waiting for CircleCI to complete, will merge then.
